src/lang/base/TypeUtil.ml

```diff
@@ -250,7 +250,10 @@ module TypeUtilities
     | ADT (_, ts) -> List.for_all ~f:(fun t -> is_storable_type t) ts
     | PolyFun _ -> false
     | Unit -> false
-    | _ -> true
+    | PrimType _ ->
+      (* Messages/Events when stored or passed in messages cannot be analyzed,
+       * leading to potential badly constructed Messages/Events. So we disallow. *)
+      not (t = PrimTypes.msg_typ || t = PrimTypes.event_typ)
```

Comment:
It's quite odd that messages and events are categorised as primitive types, now that I think about it.

I assume this will be changed as part of issue #142, in which case I'm fine with it here for now.